src/Interpreters/Streaming/benchmarks/data_blocks_perf.cpp

```diff
@@ -296,111 +296,134 @@ BENCHMARK_CAPTURE(
     Column3ChunkRows1Chunks1000,
     /*chunk_columns=*/3,
     /*chunk_rows=*/1,
-    /*chunks*=*/1'000)
+    /*chunks=*/1'000,
+    /*data_block_size=*/0)
     ->Iterations(1000);
 
 BENCHMARK_CAPTURE(
     refCountDataBlockList,
     Column3ChunkRows1Chunks10000,
     /*chunk_columns=*/3,
     /*chunk_rows=*/1,
-    /*chunks*=*/10'000)
+    /*chunks*=*/10'000,
+    /*data_block_size=*/0)
     ->Iterations(100);
 
 BENCHMARK_CAPTURE(
     refCountDataBlockList,
     Column3ChunkRows1Chunks100000,
     /*chunk_columns=*/3,
     /*chunk_rows=*/1,
-    /*chunks*=*/100'000)
+    /*chunks*=*/100'000,
+    /*data_block_size=*/0)
     ->Iterations(10);
 
 BENCHMARK_CAPTURE(
     refCountDataBlockList,
     Column3ChunkRows1Chunks1000000,
     /*chunk_columns=*/3,
     /*chunk_rows=*/1,
-    /*chunks*=*/1000'000)
+    /*chunks*=*/1000'000,
+    /*data_block_size=*/0)
     ->Iterations(10);
 
 BENCHMARK_CAPTURE(
     refCountDataBlockList,
     Column3ChunkRows1Chunks10000000,
     /*chunk_columns=*/3,
     /*chunk_rows=*/1,
-    /*chunks*=*/10'000'000)
+    /*chunks*=*/10'000'000,
+    /*data_block_size=*/0)
+    ->Iterations(1);
+
+BENCHMARK_CAPTURE(
+    refCountDataBlockList,
+    Column3ChunkRows1Chunks10000000BS1000,
+    /*chunk_columns=*/3,
+    /*chunk_rows=*/1,
+    /*chunks*=*/10'000'000,
+    /*data_block_size=*/1000)
```

Comment:
Here we only test one `data_block_size = 1000`
For rows = 1, it seems that the larger the `data_block_size`, the smaller the memory overhead.

Reply:
In general, this is true. And we can emulate other combinations in this test and there are lots of combinations. 1000 is just a good starting point for this case. And this is another reason the data block size setting is exposed to end users for now.